v3/integrations/nrfasthttp/nrfasthttp_client.go

```diff
@@ -0,0 +1,21 @@
+package nrfasthttp
+
+import (
+	newrelic "github.com/newrelic/go-agent/v3/newrelic"
+	"github.com/valyala/fasthttp"
+)
+
+type FastHTTPClient interface {
+	Do(req *fasthttp.Request, resp *fasthttp.Response) error
+}
+
+func Do(client FastHTTPClient, txn *newrelic.Transaction, req *fasthttp.Request, res *fasthttp.Response) error {
+	seg := txn.StartSegment("fasthttp-do")
+	err := client.Do(req, res)
```

Comment:
Should we inject DT headers into the request? This can get tricky. We may also want to consider filling out more info in the transaction. Like whether we are categorizing this request as a web/non-web request.